### ISSUE TITLE
Restoration can't be activated if the squad is not missing HP

### ISSUE THREAD

**Issue by copyrite** [X2CommunityCore/X2WOTCCommunityHighlander#1436]
Restoration states the following:
> The GREMLIN flies to each squad member, healing or reviving them as needed.

However, when computing whether the ability can be activated, the ability iterates through squad members. If it finds that all of them are in full health, the ability is unusable. In effect, if no one is hurt, Restoration can't be used as a comically oversized Revival Protocol.

Best also check what the interaction with any of the damaging statuses on squad members is and fix that if needed.

Originally posted to LWOTC: https://github.com/long-war-2/lwotc/issues/1727

**Comment by Iridar** [comment]
I can see the problem after looking at the code. It is indeed kinda messy. It seems the cutoff check is:

```cs
MultiTargetingStyle.NumTargetsRequired = 1; //At least someone must need healing
```
and
```cs
HealTargetCondition.ExcludeFullHealth = true;
```
So effectively the ability multi targets all units that don't have full HP and at the same time aren't dead or bleeding out. 

The build visualization function just makes the gremlin visit those targets that had any effect applied to them by the ability.

One the previous Highlander fixes might actually be messing with this intended functionality, as there is the StunRecover effect that always gets applied to all units now, so it could be it's causing the Gremlin to visit all squad units, even if it's not doing anything for them. Needs testing.

**Comment by faanlez** [comment]
`ExcludeFullHealth` actually *doesn't exclude* full health units affected by effects that are in the `MedikitHealEffectTypes` config, as this is used by the medikit/gremlin heal abilities to let you remove the statuses without needing an actual wound.

**Comment by Iridar** [comment]
![Firaxis Noodle Games](https://github.com/user-attachments/assets/1758a7ce-5ac2-4c6a-a0b8-e55e39d6191e)

**Comment by BlackDog86** [comment]
Yeah, so the targetting conditions are really wonky here - as you both say, it looks like X2Condition_UnitProperty::ExcludeFullHealth excludes (as a first priority) units that have full health unless they are effected by a status effect that can be healed by a medikit (using native logic). 

Then it manually excludes stuff which is dead or bleeding out under seperate conditions. 

My proposal is probably be to develop a custom X2Condition to replace the current mess which would allow friendly units to be targetted matching the below criteria:

Unit is not dead or bleeding out AND
(Unit has missing health OR
Unit has a status effect that can be healed with a medikit OR
Unit has a mental status effect that can be removed with Revival Protocol)

Does this sound sensible?
